lib/models.go

```diff
@@ -68,6 +70,109 @@ func (s *Stage) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+type TLSVersion struct {
+	Min int
+	Max int
+}
+
+func (v *TLSVersion) UnmarshalJSON(data []byte) error {
+	// From https://golang.org/pkg/crypto/tls/#pkg-constants
+	versionMap := map[string]int{
+		"ssl3.0": tls.VersionSSL30,
+		"tls1.0": tls.VersionTLS10,
+		"tls1.1": tls.VersionTLS11,
+		"tls1.2": tls.VersionTLS12,
+	}
+
+	// Version might be a string or an object with separate min & max fields
+	var fields struct {
+		Min string `json:"min"`
+		Max string `json:"max"`
+	}
+	if err := json.Unmarshal(data, &fields); err != nil {
+		switch err.(type) {
+		case *json.UnmarshalTypeError:
+			// Check if it's a type error and the user has passed a string
+			var version string
+			if otherErr := json.Unmarshal(data, &version); otherErr != nil {
+				// Some other error occurred or none of the types match
+				return otherErr
+			}
+			// It was a string, assign it to both min & max
+			fields.Min = version
+			fields.Max = version
+		default:
+			return err
+		}
+	}
+
+	var minVersion int
+	var maxVersion int
+	var ok bool
+	if minVersion, ok = versionMap[fields.Min]; !ok {
+		return errors.New("Unknown TLS version : " + fields.Min)
+	}
+
+	if maxVersion, ok = versionMap[fields.Max]; !ok {
+		return errors.New("Unknown TLS version : " + fields.Max)
+	}
+
+	v.Min = minVersion
+	v.Max = maxVersion
+
+	return nil
+}
+
+type TLSCipherSuites struct {
+	Values []uint16
+}
+
+func (s *TLSCipherSuites) UnmarshalJSON(data []byte) error {
+	// From https://golang.org/pkg/crypto/tls#pkg-constants
+	suiteMap := map[string]uint16{
```

Comment:
Also: I'd probably use a switch on the value instead of a map (in an unexported function); that way you can get test coverage telling you which cases are properly tested.